src/document_cleaners.py

```diff
@@ -19,3 +23,125 @@ class IdentityDocumentCleaner(BaseDocumentCleaner):
     def __call__(self, chunks: Sequence[Chunk]) -> list[Chunk]:
         """Run document cleaning"""
         return list(chunks)
+
+
+class ChunkTypeFilter(BaseDocumentCleaner):
+    """Filter out chunks of specified types."""
+
+    def __init__(self, types_to_remove: list[str]) -> None:
+        """
+        Args:
+
+        :param types_to_remove: the types of chunk to remove
+        """
+        for _type in types_to_remove:
+            try:
+                ChunkType(_type)
+            except NameError:
+                logger.warning(
+                    f"Blocks to filter should be of a known block type, removing {_type} "
+                    f"from the list. "
+                )
+                types_to_remove.remove(_type)
+
+        self.chunks_to_drop = types_to_remove
```

Comment:
any reason why the variable name changes here?

Reply:
ah whoops - a rename that i didn't propagate all the way through. good catch!